serial_driver/CMakeLists.txt

```diff
@@ -32,25 +32,55 @@ ament_auto_find_build_dependencies()
 find_package(asio_cmake_module REQUIRED)
 find_package(ASIO REQUIRED)
 
+ament_auto_add_library(${PROJECT_NAME} SHARED
+  src/serial_port.cpp
+  src/serial_driver.cpp
+)
+
+ament_target_dependencies(${PROJECT_NAME} "ASIO")
+
+ament_auto_add_library(${PROJECT_NAME}_nodes SHARED
+  src/serial_bridge_node.cpp
+)
+
+rclcpp_components_register_node(${PROJECT_NAME}_nodes
+  PLUGIN "drivers::serial_driver::SerialBridgeNode"
+  EXECUTABLE "serial_bridge_node_exe"
```

Comment:
would it be better to name this just `serial_bridge`, since you'll only really see this when calling it by `ros2 run serial_driver`, where `run` signifies an executable?

Reply:
In-general I just duplicate the class name (`SerialBridgeNode`) and add `_exe` - it's convention in Autoware.Auto too. However, if we do change this, we should change the nodes in `udp_driver` to match.

Reply:
it just seems unnecessarily long. and this name will be what people call via the command line....and usually i appreciate it if they are short 😄 this can be saved for a later time though - not really important ATM

Reply:
Shortened to just `serial_bridge`. FYI - I expect that most people will use the launch file because it automatically transitions the Lifecycle node but it's not a big deal. Done.